documentation/docs/workflow.md

```diff
@@ -0,0 +1,149 @@
+The determined version of the release is used to create the [hotfix](#hotfix) or [release](#release) branch. For example a major release from the previous version `v2.3.1` will create the `release/v3.0.0` branch.
+Notice the `v` in front of the version. We [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) all release in git in the form of `vMAJOR.MINOR.PATCH`. For bindings we prefix the tag with the binding name, so a tag for Wasm would look like `wasm-v1.2.3`. Bindings and the "core" are versioned and released independently.
+
+For bindings and and the "core" we may release `dev` versions. Those versions are meant as a preview of upcoming versions. For example if the current version is 1.2.3 with the tag `v1.2.3` we may release `v1.3.0-dev` which is then superseded by the actual `1.3.0` release.
+
+To create a release follow the following steps:
+1. Make sure the changes you want to release are on the `dev` branch
+2. Use the appropriate action, e.g. `Rust Create Release PR`
+2.1. Decide if you want to create a `dev` or `main` release.
+2.2. Determine the next version string.
+2.3. Run the workflow. The workflow will create a PR from `dev` targeting `dev` with release related changes.
+3. Review the PR
+3.1 The PR will contain an updated changelog, make sure it has all expected entries in the appropriate sections and the determined version matches the changelog according to [SemVer](#Semantic-Versioning).
+3.2 The PR will contain changes to files including versions or version strings. Make sure this matches expectations.
+4. Merge the PR
+4.1. On merge an automatic workflow is triggered that optionally creates a GitHub Release (this is determined in the workflow configuration) and builds and publishes artifacts also according to the workflow configuration.
+5. If the release is a `main` release, merge the `dev` branch into the `main` branch.
+
+### Troubleshooting
+#### The changelog in the release PR is missing entries, has unrelated entries, entries in the wrong section.
+Fix the [labels](#PR-Labels) on the related PRs and rerun the workflow with the same parameters. The PR will be updated with the updated changelog.
+#### The release description in the release PR is missing or wrong.
+Fix the issues description, milestone and label according to the [release summaries guide](#Release-Summary) and rerun the workflow with the same parameters. The PR will be updated with the updated changelog.
+#### Features / Code is missing from the release.
+Merge the code into the `dev` branch. Rerun the workflow with the same parameters.
+#### I want to abort the release for any reason.
+CLose the PR. You can reopen it later.
```

Comment:
This might need to be expanded.

```suggestion
### Troubleshooting

#### The changelog entries have the wrong description in the release PR
Update the titles of the relevant PRs then rerun the workflow with the same parameters. The release PR will be updated with the new changelog.

#### The changelog in the release PR is missing entries, has unrelated entries, entries in the wrong section.
Fix the [labels](#PR-Labels) on the relevant PRs then rerun the workflow with the same parameters. The release PR will be updated with the new changelog.

#### The release description in the release PR is missing or wrong.
Fix the issue description, milestone and label according to the [release summaries guide](#Release-Summary) and rerun the workflow with the same parameters. The release PR will be updated with the new changelog.

#### Features or code are missing from the release.
Merge the code into the `dev` branch then rerun the workflow with the same parameters. The release PR will be updated with the changes.

#### I want to abort the release for any reason.
Close the PR. You can reopen it later. // TODO: can I just re-run the workflow? Maybe that needs an "I want to resume an aborted release" section?
```